### PR TITLE
SapMachine #1124: Vitals: June 22 package (SapMachine 19)

### LINKED ISSUE #1124: Vitals: June 22 package
Update Vitals to V3

### PR DESCRIPTION
**Pull request by tstuefe**
Downport of #1124 to SapMachine 19.

Only minor adjustments had to be made, because of the recent splitup of thread.cpp, and other small fish. See c09280fb8c5baa012a1d9cd26e51dfea2fd6938c for all changes from the base patch.

fixes #1124

### CODE REVIEW

**Comment by SapMachine** [comment]
Hello @tstuefe, this pull request fulfills all formal requirements.

**Comment by tstuefe** [comment]
GHAs did run, mostly successfully. I am not sure why they are not displayed in the "Checks" pane. 

https://github.com/tstuefe/SapMachine/actions/runs/2588304666

The one error on x86 (java/lang/ref/OOMEInReferenceHandler.java) is not associated with this patch.

I am currently running debug tests too. These are still running.

**Comment by RealCLanger** [comment]
> GHAs did run, mostly successfully. I am not sure why they are not displayed in the "Checks" pane.

You could change the branch to sapmachine19 here, then you would see it (resp. they would run in the SapMachine repository):
https://github.com/tstuefe/SapMachine/blob/c09280fb8c5baa012a1d9cd26e51dfea2fd6938c/.github/workflows/submit.yml#L11

**Comment by tstuefe** [comment]
> > GHAs did run, mostly successfully. I am not sure why they are not displayed in the "Checks" pane.
> 
> You could change the branch to sapmachine19 here, then you would see it (resp. they would run in the SapMachine repository): https://github.com/tstuefe/SapMachine/blob/c09280fb8c5baa012a1d9cd26e51dfea2fd6938c/.github/workflows/submit.yml#L11

Thanks for the tip!

**Review by RealCLanger — APPROVED** [review]

File: src/hotspot/share/vitals/vitals.cpp [hunk 1]
```diff
@@ -1214,16 +1254,22 @@ void sample_jvm_values(Sample* sample, bool avoid_locking) {
...
   // Adjust VitalsSampleInterval
   if (VitalsSampleInterval == 0) {
-    log_warning(os)("Invalid VitalsSampleInterval (" UINTX_FORMAT ") specified. Vitals disabled.",
+    log_warning(vitals)("Invalid VitalsSampleInterval (" UINTX_FORMAT ") specified. Vitals disabled.",
                     VitalsSampleInterval);
     return false;
   }
 
   bool success = ColumnList::initialize();
+  success &= Legend::initialize();
```
[the remark this answers is not in the document]
Comment (tstuefe) [reply src/hotspot/share/vitals/vitals.cpp:1272]: Really? Damn. I'll check this and fix this in a follow up RFE.

[the remark this answers is not in the document]
Comment (tstuefe) [reply src/hotspot/share/vitals/vitals.cpp:1272]: Opened follow up issue so I don't forget again. https://github.com/SAP/SapMachine/issues/1162

**Pull request merged**